### DISCUSSION TITLE
How to publish back an unpublished Github page site with custom domain?

### DISCUSSION THREAD

**Discussion by UjwalKandi** [community/community#34873] [Other Feature Feedback, Questions, & Ideas]
Hi All, 

Over a year ago, I deployed my website (a private repo) with GitHub Pages. I had a premium account then, but when it expired a month ago, my account was downgraded and since then, I have been unable to make changes to my site. After multiple trials, I finally decided to unpublish the site and publish it again. Now I can't publish the site again.

Does someone know how to publish my site again?

Thanks in advance!

**Comment by SilasRodrigues19** [comment]
@UjwalKandi In ur repo u need to access ```settings``` -> ```pages``` and put your custom domain, still inside ur repo u need to have the CNAME file that must be configured in CPanel. I did it watching this video, but it's in Portuguese -> [Configurando Domínio para o GitHub Pages
](https://www.youtube.com/watch?v=51Eco4wDdvg&ab_channel=ThiagoMarinho)

Maybe u can get it by following the step by step with a caption.

**Comment by Anmol-Baranwal** [comment]
For your page to automatically be hosted, you must follow a specific naming convention. Name your repository “[your-username].github.io”, and leave everything else as is.

That’s it! Your new page should be up at http://your-username.github.io.

If you have further queries then, I think the following three resources would be sufficient: 

- check freecodecamp blog [here](https://www.freecodecamp.org/news/hosting-custom-domain-on-github-pages-8c598248d2bc/)

- check the official documentation [here](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site)
- check the official documentation on GitHub Pages [here](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)

**Note:**  You can create GitHub Pages sites that are publicly available on the internet. Organizations that use GitHub Enterprise Cloud can also publish sites privately by managing access control for the site.

**Comment by airtower-luna** [comment]
> Over a year ago, I deployed my website (a private repo) with GitHub Pages.

You can't publish a Pages site from a private repository with a free account. This is [mentioned in the docs](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages) (green box at the top):

> GitHub Pages is available in public repositories with GitHub Free and GitHub Free for organizations, and in public and private repositories with GitHub Pro, GitHub Team, GitHub Enterprise Cloud, and GitHub Enterprise Server.

You should be able to publish your site again if you make the repository public.

**Comment by K-adu** [comment]
it might be that the automated process didnt started. You can change the name of the repo once and and after a while change it back to original. For me it did the job

**Comment by eishha** [comment]
Just go back to code section of your repository. In the about section of repository, there will be a setting icon. Click on that, and you will see a option, most probably at 2nd number, of "Websit". Under website section, checkmark "Use your gihub pages website" and that's all. Your site has been published again!
![Screenshot (49)](https://github.com/community/community/assets/99905041/25b08b74-d2b4-4d53-a606-c9d981c584bf)
